cognite/client/_api/data_sets.py

```diff
@@ -0,0 +1,247 @@
+
+        Args:
+            metadata (Dict[str, str]): Custom, application-specific metadata. String key -> String value.
+            created_time (Union[Dict[str, Any], TimestampRange]): Range between two timestamps.
+            last_updated_time (Union[Dict[str, Any], TimestampRange]): Range between two timestamps.
+            external_id_prefix (str): Filter by this (case-sensitive) prefix for the external ID.
+            write_protected (bool): No description.
+            limit (int, optional): Maximum number of data sets to return. Defaults to 25. Set to -1, float("inf") or None
+                to return all items.
+
+        Returns:
+            DataSetList: List of requested data sets
+
+        Examples:
+
+            List data sets and filter on write_protected::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> data_sets_list = c.data_sets.list(limit=5, write_protected=False)
+
+            Iterate over data sets::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> for data_set in c.data_sets:
+                ...     data_set # do something with the data_set
+
+            Iterate over chunks of data sets to reduce memory load::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> for data_set_list in c.data_sets(chunk_size=2500):
+                ...     data_set_list # do something with the list
+        """
+
+        filter = DataSetFilter(
+            metadata=metadata,
+            created_time=created_time,
+            last_updated_time=last_updated_time,
+            external_id_prefix=external_id_prefix,
+            write_protected=write_protected,
+        ).dump(camel_case=True)
+        return self._list(method="POST", limit=limit, filter=filter)
+
+    def update(
+        self, item: Union[DataSet, DataSetUpdate, List[Union[DataSet, DataSetUpdate]]]
+    ) -> Union[DataSet, DataSetList]:
+        """`Update one or more data sets <https://docs.cognite.com/api/v1/#operation/updateDataSets>`_
+
+        Args:
+            item: Union[DataSet, DataSetUpdate, List[Union[DataSet, DataSetUpdate]]]: Data set(s) to update
+
+        Returns:
+            Union[DataSet, DataSetList]: Updated data set(s)
+
+        Examples:
+
+            Update an data set that you have fetched. This will perform a full update of the data set::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> data_set = c.data_sets.retrieve(id=1)
+                >>> data_set.description = "New description"
+                >>> res = c.data_sets.update(data_set)
+
+            Perform a partial update on a data set, updating the description and removing a field from metadata::
```

Comment:
One `:` too many

Reply:
it's used in every other place, I guess it's our custom syntax for code snippets